include/mitsuba/core/distr_1d_cond.h

```diff
@@ -0,0 +1,1225 @@
+#include <mitsuba/core/logger.h>
+#include <mitsuba/core/math.h>
+#include <mitsuba/core/vector.h>
+
+NAMESPACE_BEGIN(mitsuba)
+
+/**
+ *
+ * \brief Conditional 1D irregular distribution
+ *
+ *
+ * Similar to the irregular 1D distribution, but this class represents an
+ * N-Dimensional irregular one (with the extra conditional dimensions being also
+ * irregular).
+ *
+ * As an example, assume you have a 3D distribution P(x,y,z), with leading
+ * dimension X. This class would allow you to obtain the linear interpolated
+ * value of the PDF for \c x given \c y and \c z. Additionally, it allows you to
+ * sample from the distribution P(x|Y=y,Z=z) for a given \c y and \c z.
+ *
+ * It assumes every conditioned PDF has the same size.
+ *
+ * This distribution can be used in the context of spectral rendering, where
+ * each wavelength conditions the underlying distribution.
+ *
+ */
+template <typename Value> class ConditionalIrregular1D {
+    using Float        = std::conditional_t<dr::is_static_array_v<Value>,
+                                            dr::value_t<Value>, Value>;
+    using FloatStorage = DynamicBuffer<Float>;
+    using Index        = dr::uint32_array_t<Value>;
+    using UInt32       = dr::uint32_array_t<Float>;
+    using IndexFloat   = dr::uint32_array_t<Float>;
+    using Mask         = dr::mask_t<Value>;
+    using TensorXf     = dr::Tensor<FloatStorage>;
+
+    using ScalarFloat    = dr::scalar_t<Float>;
+    using ScalarVector2f = Vector<ScalarFloat, 2>;
+    using ScalarVector2u = Vector<uint32_t, 2>;
+    using ScalarUInt32   = dr::uint32_array_t<ScalarFloat>;
+
+public:
+    ConditionalIrregular1D() : m_enable_sampling(true) {};
+
+    /**
+     * \brief Construct a conditional irregular 1D distribution
+     *
+     * \param nodes
+     *      Points where the leading dimension N is defined
+     * \param pdf
+     *      Flattened array of shape [D1, D2, ..., Dn, N], containing the PDFs
+     * \param nodes_cond
+     *      Arrays containing points where each conditional dimension is
+     * evaluated
+     * \param enable_sampling
+     *      Enable sampling of the distribution
+     * \param nearest
+     *      If true, the distribution will return the nearest neighbor
+     */
+    ConditionalIrregular1D(const FloatStorage &nodes, const FloatStorage &pdf,
+                           const std::vector<FloatStorage> &nodes_cond,
+                           bool enable_sampling = true, bool nearest = false)
+        : m_nodes(nodes), m_nodes_cond(nodes_cond), m_nearest(nearest),
+          m_enable_sampling(enable_sampling) {
+
+        // Update tensor with the information
+        std::vector<size_t> shape;
+        for (size_t i = 0; i < m_nodes_cond.size(); i++)
+            shape.push_back(dr::width(m_nodes_cond[i]));
+        shape.push_back(dr::width(m_nodes));
+
+        // Initialize tensor storing the distribution values
+        m_pdf = TensorXf(pdf, shape.size(), shape.data());
```

Comment:
Wouldn't it make sense to have a constructor that just takes a `TensorXf`. I think that would feel very natural in Python.